### PR TITLE
Code Refactor

### PR DESCRIPTION
**Pull request by ManojKiranA**
PR Addresses following things

1. Log action updated event
2. Remove hard coded dropdown actions
3. Remove unused fileds from user and app query and move logic to class
4. Use loop variable property directly
5. Meaningful message when records not found
6. Use isDirty method instead of if
7. Move observer to model

I will move all the event firing logic to Model Observer in future PR 🔥

### CODE REVIEW

**Comment by ManojKiranA** [comment]
After filling the import modal if we press cancel or esc button the contents of modal is not cleared it was fixed in the commit https://github.com/envault/envault/pull/32/commits/570affb2658fdc5f97221f7944a1601222254546

**Comment by danharrin** [comment]
Awesome, thank you again @ManojKiranA!

File: resources/views/log.blade.php [hunk 1]
```diff
@@ -15,63 +15,72 @@
...
 
+                        <div class="flex mt-2 sm:mt-0 sm:mx-2 relative rounded-md shadow-sm sm:w-1/3">
                             <select wire:model="appId" @if (Str::before($this->action, '.') == 'user') disabled
                                     @endif class="@if (Str::before($this->action, '.') == 'user') bg-gray-50 opacity-50 cursor-not-allowed @endif border-gray-300 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 rounded-md block w-full transition duration-150 ease-in-out sm:text-sm sm:leading-5"
                             >
                                 <option value="">Select app...</option>
-                                @foreach ($this->auditLogApplicationList as $applicationId => $applicationName)
-                                    <option value="{{ $applicationId }}">{{ $applicationName }}</option>
+
+                                @foreach (App\Models\App::orderBy('name')->get() as $app)
```
Comment (ManojKiranA) [inline resources/views/log.blade.php:37]: we just need Application Id and Name right ??

File: resources/views/variables/create.blade.php [hunk 2]
```diff
@@ -73,11 +73,7 @@ class="fixed z-10 bottom-0 inset-x-0 px-4 pb-4 sm:inset-0 sm:flex sm:items-cente
                             <div class="absolute inset-0 bg-gray-500 opacity-75"></div>
                         </div>
 
-                        <div 
-                            @click.away="importOpen = false 
-                                        $wire.set('import', '')" 
-                            @keydown.escape.window="importOpen = false
-                                        $wire.set('import', '')"
+                        <div @click.away="importOpen = false" @keydown.escape.window="importOpen = false; $wire.set('import', '')"
```
Comment (ManojKiranA) [inline resources/views/variables/create.blade.php:76]: Re formatting to single line will break the behaviour. The conents of modal wont be cleared

**Pull request merged**